statline_bq/utils.py

```diff
@@ -470,184 +529,45 @@ def get_column_descriptions_v3(url_data_properties: str) -> dict:
-        bag = db.concat([bag, temp_bag])
-
-        target_url = r.get("@odata.nextLink", None)
-
-    return bag
-
-
-def get_odata(target_url: str, odata_version: str) -> db:
-    """Gets a table from a valid CBS url and returns it as a Dask bag.
-
-    A wrapper, calling the appropriate version function to get the table from
-    a valid CBS url, leading to a table in OData format.
-
-    Parameters
-    ----------
-    target_url : str
-        A url to the table
-    odata_version : str
-        version of the odata for this dataset - must be either "v3" or "v4".
-
-    Returns
-    -------
-    Dask Bag
-        All data received from target url as json type, concatenated in a Dask Bag
-
-    Raises
-    ------
-    ValueError
-        If 'odata_version` is not one of {"v3", "v4"}
-    """
-
-    if odata_version == "v4":
-        return get_odata_v4(target_url)
-    elif odata_version == "v3":
-        return get_odata_v3(target_url)
-    else:
-        raise ValueError("odata version must be either 'v3' or 'v4'")
-
-
-def get_odata_v3(target_url: str) -> db:
-    """Gets a table from a valid url for CBS dataset with Odata v3.
-
-    This function uses standard requests.get() to retrieve data at target_url
-    in json format, and concats it all into a Dask Bag to handle memory
-    overflow if needed.
-
-    Each request from CBS is limited to 10,000 rows, and if more data exists
-    the key "odata.nextLink" exists in the response with the link to the next
-    10,000 (or less) rows.
-
-    Meant to be wrapped by `get_odata()`, and it is better practice to call it
-    wrapped to allow for both "v3" and "v4" functionality.
-
-    Parameters
-    ----------
-    target_url: str
-        A valid url of a table from CBS
-
-    Returns
-    -------
-    bag: Dask Bag
-        All data received from target url as json type, concatenated as a Dask bag
-    """
-
-    print(f"Fetching from {target_url}")
-    # First call target url and get json formatted response as dict
-    r = requests.get(target_url).json()
-
-    # Initialize bag as None
-    bag = None
-
-    # Create Dask bag from dict (check if not empty field)
-    if r["value"]:
-        bag = db.from_sequence(r["value"])  # TODO -> define npartitions?
-
-    # check if more data exists
-    target_url = r.get("odata.nextLink", None)
-
-    # if more data exists continue to concat bag until complete
-    while target_url:
-        r = requests.get(target_url).json()
-        if r["value"]:
-            temp_bag = db.from_sequence(r["value"])
-            bag = db.concat([bag, temp_bag])
-
-        target_url = r.get("odata.nextLink", None)
-
-    return bag
-
-
-def get_odata_v4(target_url: str,) -> db:
-    """Gets a table from a specific url for CBS Odata v4.
-    This function uses standard requests.get() to retrieve data at target_url
-    in json format, and concats it all into a Dask Bag to handle memory
-    overflow if needed.
+def get_odata(table_urls: list) -> db:
```

Comment:
`db` is not a Python type, but an alias. You probably want to write this for proper type-hinting:

```
def get_odata(table_urls: list) -> dask.bag.Bag:
...
```

Reply:
No problem, but just fyi - that requires importing the whole dask library using `import dask` in addition to the currentto `import dask.bag as db` - I am not sure that makes sense to me.

Reply:
You're mixing up a couple of things:

1. ` import dask.bag  as db` imports the whole module.
2. ` from dask.bag import Bag` imports the Bag class
3. ` import dask.bag.Bag as db` imports the Bag class with an alias `db`

For type hinting you can use the 2nd or 3rd option. Just looked in up, apparently it's OK to use aliases.
https://docs.python.org/3/library/typing.html

Reply:
ah, I see. Well, the 3rd options throws a `ModuleNotFoundError` error, so I'll use the 2nd form: `from dask.bag import Bag as DaskBag` for type hinting, and keep the 1st as it is now for the actual module.